pkg/collector/runner/runner.go

```diff
@@ -43,24 +52,36 @@ func init() {
 
 // Runner ...
 type Runner struct {
-	pending       chan check.Check         // The channel where checks come from
-	done          chan bool                // Guard for the main loop
-	runningChecks map[check.ID]check.Check // the list of checks running
-	m             sync.Mutex               // to control races on runningChecks
-	running       uint32                   // Flag to see if the Runner is, well, running
+	pending          chan check.Check         // The channel where checks come from
+	done             chan bool                // Guard for the main loop
+	runningChecks    map[check.ID]check.Check // The list of checks running
+	m                sync.Mutex               // To control races on runningChecks
+	running          uint32                   // Flag to see if the Runner is, well, running
+	staticNumWorkers bool                     // Flag indicating if numWorkers is dynamically updated
 }
 
 // NewRunner takes the number of desired goroutines processing incoming checks.
-func NewRunner(numWorkers int) *Runner {
+func NewRunner() *Runner {
+	numWorkers := config.Datadog.GetInt("check_runners") // = 0 if no value is specified by the user
+	if numWorkers > maxNumWorkers {
```

Comment:
we'd want to log something here if the configured value will not be enforced because it's higher than the `maxNumWorkers`

Reply:
✅